modules/image-loading-optimization/assets/js/editor/hooks/check-lcp-image-before-save.js

```diff
@@ -0,0 +1,138 @@
+export class CheckLCPImageBeforeSave extends $e.modules.hookData.Dependency {
+	getCommand() {
+        return 'document/save/save';
+    }
+
+    getId() {
+        return 'check-lcp-image-before-save';
+    }
+
+    getConditions( args ) {
+        return 'publish' === args.status && ! args.force; // Don't check if it's a forced save.
+    }
+
+   async apply() {
+        const currentDeviceMode = this.getCurrentDeviceMode(),
+        currentScrollPosition = this.getPreviewFrame().scrollY;
+
+        // Switch to desktop mode to get the LCP image.
+        const deviceMode = 'desktop'
```

Comment:
@joemcgill  Currently only supports only desktop. I did try switching to other device mode but it was not capturing new LCP even after page reload inside iframe using `elementor.reloadPreview();` possible issue with `DOMContentLoaded` only called once ( [this line ](https://github.com/kt-12/elementor/pull/3/files#diff-85e71fb51b8e4fbbd1b086e2e213f371f627ab0b9cdf486af9b5f2f3495282f8R85) )